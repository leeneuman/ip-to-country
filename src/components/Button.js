import React from 'react';

function Button({ label, onClick }) {
	return (
		<button type="button" className="btn btn-primary mt-3" onClick={onClick} aria-label={label}>
			<span className="mr-1" aria-hidden="true">
				&#x2B;
			</span>
			{label}
		</button>
	);
}

export default Button;
