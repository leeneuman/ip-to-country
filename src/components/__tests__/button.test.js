import { render, screen, cleanup, fireEvent } from '@testing-library/react';
import renderer from 'react-test-renderer';
import Button from './../button';

afterEach(cleanup);

test('should render button component', () => {
	render(<Button label="test label" />);
	const buttonElement = screen.getByTestId('button');
	expect(buttonElement).toBeInTheDocument();
	expect(buttonElement).toHaveTextContent('test label');
});

test('matches snapshot', () => {
	const tree = renderer.create(<Button label="test label" />).toJSON();
	expect(tree).toMatchSnapshot();
});

test('calls the onClick function', () => {
	const onClick = jest.fn();
	render(<Button onClick={onClick} />);
	const buttonElement = screen.getByTestId('button');
	fireEvent.click(buttonElement);
	expect(onClick).toHaveBeenCalled();
});
