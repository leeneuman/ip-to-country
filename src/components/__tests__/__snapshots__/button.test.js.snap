// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<button
  aria-label="test label"
  className="btn btn-primary mt-3"
  data-testid="button"
  type="button"
>
  <span
    aria-hidden="true"
    className="mr-1"
  >
    +
  </span>
  test label
</button>
`;
